packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -600,8 +600,15 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
     return rows;
   };
 
+  let offset = 0;
+
+  const current = (ref as unknown as React.MutableRefObject<HTMLButtonElement | null>).current;
+  if (current) {
+    offset = current.offsetWidth - current.clientWidth;
+  }
```

Comment:
```suggestion
  const scrollBarWidth = (rootRef?.current?.offsetWidth || 0) - (rootRef?.current?.clientWidth || 0);
```

To make it clear that we're calculating scrollbar width here?

Reply:
I think it's for the scrollbar arrow. should I name it `scrollBarArrowWidth`

![image](https://user-images.githubusercontent.com/34584266/231683568-02a072a3-8255-4bb2-a6f0-2c4a0bb83ffd.png)

Reply:
> Do you think you could cover it with a unit test?

sure, but I have some questions about it. can I run the layout test on locally or do I have to push it?

Reply:
> I think it's for the scrollbar arrow. should I name it scrollBarArrowWidth

I don't think it's a scrollbar arrow width:

> `offsetWidth`
> offsetWidth is a measurement in pixels of the element's CSS width, **including any borders, padding, and vertical scrollbars** (if rendered)
> https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetWidth

> `clientWidth`
> It includes padding but **excludes borders, margins, and vertical scrollbars** (if present).
> https://developer.mozilla.org/en-US/docs/Web/API/Element/clientWidth

In this case, the element doesn't have borders or padding, so `offsetWidth - clientWidth` gives us the vertical scrollbar width, right?

Reply:
> sure, but I have some questions about it. can I run the layout test on locally or do I have to push it?

Yes, you can run unit tests in the browser with `yarn test:karma`. To run only a subset of tests - use `it.only(` or `describe.only(`
Let me know if I missed something

Reply:
> In this case, the element doesn't have borders or padding, so offsetWidth - clientWidth gives us the vertical scrollbar width, right?

yea, you are right. I thought the horizontal scrollbar was causing the `width` to change. I have not realized the width is also affected by `needsHorizontalScrollbar`.

> Let me know if I missed something

I am having trouble coming up with a test. I could not write a failing test. it's kind of tricky b/c I need to check the grid is resizing continuously.